recipes/cspice/all/conanfile.py

```diff
@@ -0,0 +1,160 @@
+import os
+import time
+
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+
+class CspiceConan(ConanFile):
+    name = "cspice"
+    description = "NASA C SPICE library"
+    license = "TSPA"
+    topics = ("conan", "spice", "naif", "kernels", "space", "nasa", "jpl", "spacecraft", "planet", "robotics")
+    homepage = "https://naif.jpl.nasa.gov/naif/toolkit.html"
+    url = "https://github.com/conan-io/conan-center-index"
+    exports_sources = ["CMakeLists.txt", "patches/**"]
+    exports = ["TSPA.txt"]
+    generators = "cmake"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {"shared": False, "fPIC": True}
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    @property
+    def _patch_per_platform(self):
+        return {
+            "Macos": {
+                "x86": {
+                    "apple-clang": ["to-MacIntel-OSX-or-SunSPARC-Solaris.patch", "to-MacIntel-OSX-AppleC-32bit.patch"]
+                },
+                "x86_64": {
+                    "apple-clang": ["to-MacIntel-OSX-or-SunSPARC-Solaris.patch", "to-MacIntel-OSX-AppleC-64bit.patch"]
+                }
+            },
+            "Linux": {
+                "x86": {
+                    "gcc": ["to-PC-Linux-GCC-32bit.patch"]
+                },
+                "x86_64": {
+                    "gcc": ["to-PC-Linux-GCC-64bit.patch"]
+                }
```

Comment:
are you sure it doesn't work with CLang?

Reply:
Im' pretty sure it will compile with clang on Linux, and probably provide the expected behaviour and accuracy.
But:
- it is not officially supported by cspice. I quote what they in their README:

> We also strongly recommend that you not try to port this package to any
   other environment without first consulting NAIF. By saying "porting" we
   mean trying to build this package on a different computer hardware or
   operating system type, or using a different compiler brand or a
   different version of the compiler used to build this package, or
   changing any compile options, such as optimization level, that are used
   in the script(s) building the libraries included in this package.

- cspice source code is generated from fortran by f2c, I don't know which side effect may occur by naively trying to support an other os/arch/compiler.
- there is one define specific to each os/arch/compiler, and it is defined in a public header. It controls several underlying types of cspice types. Of course we can try to reuse an existing define (it will just be strange to have CSPICE_PC_LINUX_64BIT_GCC defined even though is was compiled with clang).
- one function prints a hard coded message depending on os/arch/compiler. I don't think it has other effects, but it will be also very strange.

Reply:
Thanks for clarifying!